utils/text_formatting.tsx

```diff
@@ -202,16 +369,23 @@ export function autolinkAtMentions(text, tokens) {
     return output;
 }
 
-function autolinkChannelMentions(text, tokens, channelNamesMap, team) {
-    function channelMentionExists(c) {
+function autolinkChannelMentions(
+    text: string,
+    tokens: Tokens,
+    channelNamesMap: ChannelNamesMap,
+    team?: Team
+) {
+    function channelMentionExists(c: string) {
         return Boolean(channelNamesMap[c]);
     }
-    function addToken(channelName, mention, displayName) {
+    function addToken(channelName: string, mention: string, displayName: string) {
         const index = tokens.size;
         const alias = `$MM_CHANNELMENTION${index}$`;
         let href = '#';
         if (team) {
-            href = (window.basename || '') + '/' + team.name + '/channels/' + channelName;
+            href =
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        ((window as any).basename || '') + '/' + team.name + '/channels/' + channelName;
```

Comment:
Do you mind joining this onto one line so that the comment can be moved to the line above it? It's kind of hard to read this with a comment in the middle of it.

Also, I'll file a ticket to figure out how to type the window object correctly since we've added that field to it, or to just get rid of that field entirely